### PR TITLE
UIREQ-939 JEST/RTL test cases for PickupServicePointFilter

### PR DESCRIPTION
**Pull request by KetineniM**
UIREQ-939 JEST/RTL test cases for PickupServicePointFilter
URL: https://issues.folio.org/browse/UIREQ-939

### CODE REVIEW

**Review by cchitneedi — APPROVED** [review]

File: src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      <div><button type="button" onClick={() => onClearFilter()} data-testid={`clear-${name}`}>Clear</button></div>
+      <div data-testid={`accordion-${name}`}>{children}</div>
+    </div>
+  )),
+  FilterAccordionHeader: jest.fn(() => <div>Filter Accordion Header</div>),
+}));
+
+jest.mock('@folio/stripes/smart-components', () => ({
+  MultiSelectionFilter: jest.fn(() => <div>Multi Selection Filter</div>)
+}));
```
Comment (artem-blazhko) [inline src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js:17]: Could you add mocks to appropriate files here if they does not exist yet - `test/jest/__mock__` ?

File: src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import PickupServicePointFilter from './PickupServicePointFilter';
+
+jest.mock('@folio/stripes/components', () => ({
+  Accordion: jest.fn(({ name, children, onClearFilter }) => (
+    <div>
+      <div><button type="button" onClick={() => onClearFilter()} data-testid={`clear-${name}`}>Clear</button></div>
```
Comment (artem-blazhko) [inline src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js:8]:
```suggestion
      <div>
        <button
          type="button"
          onClick={onClearFilter}
          data-testid={`clear-${name}`}
        >Clear</button>
      </div>
```

File: src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        activeValues={[]}
+        servicePoints={servicePoints}
+        onChange={onChange}
+        onClear={onClear}
+      />
+    );
+    const pickupServicePointsButton = screen.getByTestId('clear-pickupServicePoints');
+    expect(pickupServicePointsButton).toBeInTheDocument();
+    userEvent.click(pickupServicePointsButton);
+    expect(onClear).toHaveBeenCalled();
```
Comment (artem-blazhko) [inline src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js:44]: Could you add two separate tests for each expect?

File: src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+];
+
+describe('PickupServicePointFilter', () => {
+  const onChange = jest.fn();
+  const onClear = jest.fn();
+  beforeEach(() => {
+    onChange.mockClear();
+    onClear.mockClear();
+  });
+  it('should render the component with default props', () => {
```
Comment (artem-blazhko) [inline src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js:32]: The name of the test does not fit test scenario.

**Review by artem-blazhko — CHANGES_REQUESTED** [review]
Could you add appropriate changes to changelog file?

File: CHANGELOG.md [hunk 5]
```diff
@@ -24,6 +24,8 @@
 * Added requestDate token. Refs UIREQ-962.
 * Update `request-storage` okapi interface to `6.0` version. Refs UIREQ-963.
 * UI tests replacement with RTL/Jest for src/PatronBlockModal.js. Refs UIREQ-878.
+* Create Jest/RTL test for PickupServicePointFilter.js Refs UIREQ-939.
+
```
Comment (artem-blazhko) [inline CHANGELOG.md:28]: Could you remove empty line?

File: src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import userEvent from '@testing-library/user-event';
+import PickupServicePointFilter from './PickupServicePointFilter';
+
+const servicePoints = [
+  { id: '1', name: 'Service Point 1' },
+  { id: '2', name: 'Service Point 2' },
+  { id: '3', name: 'Service Point 3' },
+];
+
+describe('PickupServicePointFilter', () => {
```
Comment (artem-blazhko) [inline src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js:12]: Could you also check that `MultiSelectionFilter` is rendered and called with correct props?

File: src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      <PickupServicePointFilter
+        activeValues={[]}
+        servicePoints={servicePoints}
+        onChange={onChange}
+        onClear={onClear}
+      />
+    );
+    const pickupServicePointsButton = screen.getByTestId('clear-pickupServicePoints');
+    userEvent.click(pickupServicePointsButton);
+    expect(onClear).toHaveBeenCalled();
```
Comment (artem-blazhko) [inline src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js:30]: Would be great to check that it was called with correct argument.

File: test/jest/__mock__/stripesComponents.mock.js [hunk 8]
```diff
@@ -2,14 +2,29 @@ import React from 'react';
...
+          onClick={() => onClearFilter()}
+          data-testid={`clear-${name}`}
+        >Clear
+        </button>
+      </div>
+      <div data-testid={`accordion-${name}`} />
     </div>
   )),
-  AccordionSet: jest.fn(({ children }) => (
+  AccordionSet: jest.fn(({ children, onToggle }) => (
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesComponents.mock.js:20]: I do not see the usage of this component inside `PickupServicePointFilter`. Do we really need to mock it?

File: test/jest/__mock__/stripesComponents.mock.js [hunk 9]
```diff
@@ -33,9 +48,23 @@ jest.mock('@folio/stripes/components', () => ({
       {children}
     </div>
   )),
-  ConfirmationModal: jest.fn(() => <div>ConfirmationModal</div>),
+  ConfirmationModal: jest.fn(({ heading, confirmLabel, cancelLabel, onConfirm, onCancel }) => (
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesComponents.mock.js:51]: I do not see the usage of this component inside `PickupServicePointFilter`. Do we really need to mock it?

File: test/jest/__mock__/stripesComponents.mock.js [hunk 10]
```diff
@@ -33,9 +48,23 @@ jest.mock('@folio/stripes/components', () => ({
...
+      <span>ConfirmationModal</span>
+      <div>
+        <div>{heading}</div>
+        <button type="button" onClick={onConfirm}>{confirmLabel}</button>
+        <button type="button" onClick={onCancel}>{cancelLabel}</button>
+      </div>
+    </div>)),
   Datepicker: jest.fn(() => <div>Datepicker</div>),
-  ErrorModal: jest.fn(() => <div>ErrorModal</div>),
+  ErrorModal: jest.fn(({ label, content, buttonLabel, onClose }) => (
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesComponents.mock.js:61]: I do not see the usage of this component inside `PickupServicePointFilter`. Do we really need to mock it?

File: test/jest/__mock__/stripesComponents.mock.js [hunk 11]
```diff
@@ -122,7 +151,24 @@ jest.mock('@folio/stripes/components', () => ({
       {children}
     </div>
   )),
-  Select: jest.fn(() => <div>Select</div>),
+  Select: ({ name, label, onChange, required, children, value }) => {
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesComponents.mock.js:154]: I do not see the usage of this component inside `PickupServicePointFilter`. Do we really need to mock it?

File: test/jest/__mock__/stripesSmartComponents.mock.js [hunk 12]
```diff
@@ -4,8 +4,18 @@ jest.mock('@folio/stripes/smart-components', () => ({
...
+        <div>NoteCreatePage</div>
+        {props.renderReferredRecord()}
+      </div>
+    );
+  },
   NotesSmartAccordion: jest.fn(() => null),
   SearchAndSort: jest.fn(() => null),
   ViewMetaData: jest.fn(() => null),
   withTags: jest.fn((WrappedComponent) => (props) => <WrappedComponent {...props} />),
+  ProxyManager: () => (<div data-testid="proxy-manager" />),
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesSmartComponents.mock.js:19]: I do not see the usage of this component inside `PickupServicePointFilter`. Do we really need to mock it?

File: test/jest/__mock__/stripesSmartComponents.mock.js [hunk 13]
```diff
@@ -4,8 +4,18 @@ jest.mock('@folio/stripes/smart-components', () => ({
   ClipCopy: jest.fn(() => null),
   makeQueryFunction: jest.fn((value) => value),
   CheckboxFilter: jest.fn(() => null),
+  NoteCreatePage: (props) => {
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesSmartComponents.mock.js:7]: I do not see the usage of this component inside PickupServicePointFilter. Do we really need to mock it?

**Review by Dmitriy-Litvinenko — COMMENTED** [review]
Could you please fix feedbacks?

**Comment by KetineniM** [comment]
> Could you please fix feedbacks?

I have fixed the feedback issues, could you please check and let me know

**Comment by Dmitriy-Litvinenko** [comment]
Could you please fix feedbacks?

**Comment by KetineniM** [comment]
> Could you please fix feedbacks?

I have made all the possible changes. Please let me know if there is anything else remaining.

File: CHANGELOG.md (hunk 5 above)
Comment (artem-blazhko) [reply CHANGELOG.md:28]: Could you remove empty line?

File: test/jest/__mock__/stripesComponents.mock.js [hunk 14]
```diff
@@ -2,10 +2,19 @@ import React from 'react';
...
+      <div>
+        <button
+          type="button"
+          onClick={onClearFilter}
+          data-testid={`clear-${name}`}
+        >Clear
+        </button>
+      </div>
+      <div data-testid={`accordion-${name}`} />
     </div>
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesComponents.mock.js:18]:
```suggestion
  Accordion: jest.fn(({ children, label, name, onClearFilter }) => (
    <div data-testid={`accordion-${name}`} >
      {label}
      {children}
      <div>
        <button
          type="button"
          onClick={onClearFilter}
          data-testid={`clear-${name}`}
        >Clear
        </button>
      </div>
    </div>
```

File: test/jest/__mock__/stripesSmartComponents.mock.js (hunk 13 above)
Comment (artem-blazhko) [reply test/jest/__mock__/stripesSmartComponents.mock.js:7]: I do not see the usage of this component inside PickupServicePointFilter. Do we really need to mock it?

File: src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+    onChange.mockClear();
+    onClear.mockClear();
+    setupPickupServicePointFilter();
+  });
+  it('should render MultiSelectionFilter', () => {
+    const MultiSelectionFilter = screen.getByText('MultiSelectionFilter');
+    expect(MultiSelectionFilter).toBeInTheDocument();
+  });
+  it('MultiSelectionFilter should render with activeValues', () => {
+    const activeValue = screen.getByText('test');
```
Comment (artem-blazhko) [inline src/components/RequestsFilters/PickupServicePointFilter/PickupServicePointFilter.test.js:35]:
```suggestion
    const activeValue = screen.getByText(activeValues[0]);
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -27,6 +27,7 @@
 * Added requestDate token. Refs UIREQ-962.
 * Update `request-storage` okapi interface to `6.0` version. Refs UIREQ-963.
 * UI tests replacement with RTL/Jest for src/PatronBlockModal.js. Refs UIREQ-878.
+* Create Jest/RTL test for PickupServicePointFilter.js Refs UIREQ-939.
```
Comment (artem-blazhko) [inline CHANGELOG.md:30]: Could you move it to the bottom of the list?

**Pull request merged**